lib/facter/framework/config/fact_groups.rb

```diff
@@ -13,6 +13,7 @@ def initialize(group_list_path = nil)
       @groups_file_path = group_list_path || default_path
       @groups ||= File.readable?(@groups_file_path) ? Hocon.load(@groups_file_path) : {}
       load_groups
+      load_groups_from_options
```

Comment:
I think more meaningful names are required. Something like load_core_groups and load_external_and_custom_groups.